feature-libs/checkout/scheduled-replenishment/components/services/checkout-replenishment-form-service.ts

```diff
@@ -25,17 +44,74 @@ export class CheckoutReplenishmentFormService {
   private scheduleReplenishmentFormData$: BehaviorSubject<ScheduleReplenishmentForm> =
     new BehaviorSubject<ScheduleReplenishmentForm>(this.defaultFormData);
 
-  constructor() {}
+  protected orderType$ = new BehaviorSubject<ORDER_TYPE>(
+    ORDER_TYPE.PLACE_ORDER
+  );
+
+  constructor(protected eventService: EventService) {
+    this.registerOrderTypeEventListers();
+  }
+
+  protected registerOrderTypeEventListers(): void {
+    this.subscriptions.add(
+      merge(
+        this.eventService.get(DeliveryAddressSetEvent),
+        this.eventService.get(LogoutEvent),
+        this.eventService.get(LoginEvent),
+        this.eventService.get(DeliveryAddressClearedEvent),
+        this.eventService.get(DeliveryModeSetEvent),
+        this.eventService.get(DeliveryModeClearedEvent),
+        this.eventService.get(SaveCartSuccessEvent),
+        this.eventService.get(RestoreSavedCartSuccessEvent),
+        this.eventService.get(PaymentDetailsCreatedEvent),
+        this.eventService.get(PaymentDetailsSetEvent),
+        this.eventService.get(OrderPlacedEvent),
+        this.eventService.get(ReplenishmentOrderScheduledEvent),
+        this.eventService.get(MergeCartSuccessEvent)
+      ).subscribe(() => {
+        this.orderType$.next(ORDER_TYPE.PLACE_ORDER);
+      })
+    );
+  }
 
+  /**
+   * Get replenishment form data
+   */
   getScheduleReplenishmentFormData(): Observable<ScheduleReplenishmentForm> {
     return this.scheduleReplenishmentFormData$.asObservable();
   }
 
+  /**
+   * Set replenishment form data
+   * @param formData : an object containing the data for scheduling a replenishment order
+   */
   setScheduleReplenishmentFormData(formData: ScheduleReplenishmentForm): void {
     this.scheduleReplenishmentFormData$.next(formData);
   }
 
+  /**
+   * Clears the existing replenishment form data to include the default replenishment form data
+   */
   resetScheduleReplenishmentFormData(): void {
     this.scheduleReplenishmentFormData$.next(this.defaultFormData);
   }
+
+  /**
+   * Get current checkout order type
+   */
+  getOrderType(): Observable<ORDER_TYPE> {
+    return this.orderType$.asObservable();
```

Comment:
We can now just return it like this, and reduce the bundle size a bit ;)
```suggestion
    return this.orderType$;
```

Reply:
I actually checked the relevant bundles:
- `umd` (used by Node): dist/checkout/bundles/spartacus-checkout-scheduled-replenishment-components.umd.js - OLD _61 544 bytes_ vs. NEW _61 529 bytes_
- `fesm2015` (used by Bundlers / modern browsers): dist/checkout/fesm2015/spartacus-checkout-scheduled-replenishment-components.js - OLD _38 596 bytes_ vs NEW _38 581 bytes_

Reply:
coolio, didn't know